### PR TITLE
Update hal functions to match headers

### PR DESCRIPTION
**Pull request by Twinters007**

### CODE REVIEW

**Comment by Twinters007** [comment]
It looks like this was the only thing to update...

**Comment by virtuald** [comment]
But @ArthurAllshire said there were functions he was missing?

**Comment by Twinters007** [comment]
he did say that didn't he...hmmm

File: hal-sim/hal_impl/functions.py [hunk 1]
```diff
@@ -1023,8 +1035,7 @@ def getMatchTime(status):
         return (hooks.getFPGATime() - hal_data['time']['match_start'])/1000000.0
 
 def waitForDSData():
-    with hooks.ds_cond:
-        hooks.ds_cond.wait()
+    waitForDSDataTimeout(0)
```
Comment (virtuald) [inline hal-sim/hal_impl/functions.py]: I think this should be `None`

File: hal-sim/hal_impl/functions.py [hunk 2]
```diff
@@ -243,6 +243,12 @@ def getBrownedOut(status):
     status.value = 0
     return False
 
+def baseInitialize(status):
```
Comment (virtuald) [inline hal-sim/hal_impl/functions.py]: Hm, shouldn't this call initialize()? Why are there two of them now?

File: hal-sim/hal_impl/functions.py [hunk 3]
```diff
@@ -1044,6 +1055,19 @@ def observeUserProgramTeleop():
 def observeUserProgramTest():
     hal_data['user_program_state'] = 'test'
 
+def releaseDSMutex():
```
Comment (virtuald) [inline hal-sim/hal_impl/functions.py]: This is a bit strange, why isn't there a corresponding acquire function? What is the original function supposed to do (did you read the original source?)

File: hal-sim/hal_impl/functions.py [hunk 4]
```diff
@@ -1044,6 +1055,19 @@ def observeUserProgramTeleop():
...
+    hooks.ds_cond.release()
+
+def isNewControlData():
+    with hooks.ds_cond:
+        return True
```
Comment (virtuald) [inline hal-sim/hal_impl/functions.py]: This also seems wrong.

File: hal-sim/hal_impl/functions.py (hunk 1 above)
Comment (Twinters007) [reply hal-sim/hal_impl/functions.py]: the variable should be?

Comment (virtuald) [reply hal-sim/hal_impl/functions.py]: 0 should be None, as None is an infinite timeout... I don't know if 0 is.

File: hal-sim/hal_impl/functions.py (hunk 3 above)
Comment (Twinters007) [reply hal-sim/hal_impl/functions.py]: Couldn't tell you...

File: hal-sim/hal_impl/functions.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 from hal_impl.sim_hooks import SimHooks
 
 import logging
+from hal_impl.mode_helpers import notify_new_ds_data
```
Comment (auscompgeek) [inline hal-sim/hal_impl/functions.py]: This creates an import loop. This completely breaks on Python 3.4.

Comment (Twinters007) [reply hal-sim/hal_impl/functions.py]: Ahh that’s why that broke

**Comment by virtuald** [comment]
Looks like the headers/libraries are being pushed again: http://first.wpi.edu/FRC/roborio/maven/development/edu/wpi/first/hal/hal/2018.1.1-beta-1-20171018005731-8-g2fc6068/

There's a separate file for headers and for the library though, so we'll have to update the code that downloads them now.

**Comment by virtuald** [comment]
I'm going to grab this PR and do it today sometime I think, since I suspect it'll end up blocking other PRs.

**Comment by virtuald** [comment]
Closed in favor of #294

**Pull request closed without merging**